### ISSUE TITLE
Function for flexible access to the content of a saver / checkpoint file

### ISSUE THREAD

**Issue by jerabaul29** [tensorflow/tensorflow#1669]
As far as I am aware, the only ways to interact with a saver file through API are (from https://www.tensorflow.org/versions/r0.7/how_tos/variables/index.html , I also looked at the API documentation online):
- to restore a full model into exactly the same computational graph
- to restore variables 'one at a time' through an op.
Am I right or did I miss something?

Would it be possible to release additional API functions giving more flexible access to a saver file? Like making it possible to:
- list all tensors saved and their size, name etc contained in a saver file
- access a tensor saved in a saver file by just reading its content and outputting it as a series of numpy arrays (for easy post processing), instead of needing to add an op, run it etc.

**Comment by keveman** [comment]
2966fcd adds support for most of what you want. Assigning to @sherrym for further updates.

**Comment by sherrym** [comment]
286c5a4 added support to print the content of a tensor.

**Comment by sherrym** [comment]
c85b942 added a python-based tool for inspecting checkpoint.

To print shape information:
  inspect_checkpoint --file_name=model.ckpt-4971

To print contents of a particular tensor:
  inspect_checkpoint --file_name=model.ckpt-4971 \
    --tensor_name=mixed_17x17x768e/branch7x7dbl/Conv_2/weights/ExponentialMovingAverage

NOTE: If you see this message, it's likely that your checkpoint file
has been compressed with SNAPPY:

  "Data loss: Unable to open table file /tmp/model.ckpt-4971: Data
  loss: corrupted compressed block contents: perhaps your file is in a
  different file format and you need to use a different restore operator?
  It's likely that your checkpoint file has been compressed with SNAPPY."

To work around this issue without regenerating the
checkpoint files for now, you can rebuild by defining SNAPPY in
tensorflow/core/platform/posix/port.cc and link "lib_internal" with snappy
(-lsnappy) in tensorflow/core/BUILD.

**Comment by petteriTeikari** [comment]
I got the same error with the "Data loss, etc.", but could not readily figure out how this could be fixed @sherrym ?

I mean fine-tuning from `inception_resnet_v2_2016_08_30.ckpt` works using TF-Slim. The evaluation of the fine-tuned model works too, but when I try to do inference using the newly created checkpoint I got the same error (same with `inspect_checkpoint`)

The demo inference from there works fine as well: [https://github.com/tensorflow/models/blob/master/slim/slim_walkthough.ipynb](https://github.com/tensorflow/models/blob/master/slim/slim_walkthough.ipynb)
